crates/ruff/src/registry.rs

```diff
@@ -880,6 +881,18 @@ impl Rule {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiagnosticKind {
+    /// The identifier of the corresponding [`Rule`].
+    pub rule: String,
```

Comment:
I kind of wish there were `&'static str`, but it causes problems with `Deserialize`.